client/src/charts/extended_canada.js

```diff
@@ -0,0 +1,217 @@
+
+import {
+  formats,
+  run_template,
+  businessConstants,
+  declarative_charts,
+  NivoResponsiveHBar,
+  hex_to_rgb,
+  Canada,
+} from "../panels/shared.js"; 
+import { Fragment } from 'react';
+import { trivial_text_maker } from '../models/text.js';
+
+const { GraphLegend } = declarative_charts;
+const { provinces } = businessConstants;
+
+const graph_color = window.infobase_color_constants.secondaryColor;
+
+const get_graph_color = (alpha) => {
+  const rgb = hex_to_rgb(graph_color);
+  return rgb && `rgba(${rgb.r}, ${rgb.g}, ${rgb.b}, ${alpha || 1})`;
+};
+
+const format_prov_data = (prov, years_by_province, context_years) => {
+  var prov_data;
+  if(prov==="Canada"){
```

Comment:
A bit weird to me that "Canada" is a special `prov` case. I feel like it makes more sense to have `prov` be false/null when nothing's hovered, with everything set up so that the Canada case is the default/fallback. Then instead of checking `prov==="Canada"` in multiple places you just need to do a single check for if `prov` has a truthy of falsy value before either using defaults or applying the general province logic.